guggero/lightning-terminal
# Add manual renew account flow to the UI

Added a "Renew" button to the account section of the Pool page. It currently is only displayed when the active account is expired or will expire in less than 3 days.

Note: I cherry picked commits https://github.com/guggero/lightning-terminal/commit/a1d2c4e68a0d1c7294152f7f0472391b3ab9653e and https://github.com/guggero/lightning-terminal/commit/69e05a665bb858a9171bbba46c63d86d0a785d90 from the [`lnd-12`](https://github.com/lightninglabs/lightning-terminal/tree/lnd-12) branch in order to use the `RenewAccount` RPC which was added in `pool` v0.4.1. We should hold off on merging this PR until `lnd` v0.12.0 is released and those commits can be merged into the `pool-ui` branch. I will rebase this PR when that happens so it only contains the frontend changes.

### Steps to test:
1. create an account which expires in less than 432 blocks
2. click on the **Renew** button in the account section
3. enter the new expiration and fee rate
4. click the **Renew** button
5. click the **Confirm** button
6. Verify that the account has been renewed
7. Mine enough blocks to make the account expire
8. click on the **Renew** button
9. Repeat steps 3-6

Closes #46
